go-controller/pkg/node/gateway_iptables.go

```diff
@@ -364,7 +364,8 @@ func getLocalGatewayNATRules(ifname string, cidr *net.IPNet) []iptRule {
 			table: "filter",
 			chain: "FORWARD",
 			args: []string{
-				"-i", ifname,
+				"-o", ifname,
```

Comment:
so I don't get why these things are changing? why are you swapping out:
```
		{
			table: "filter",
			chain: "FORWARD",
			args: []string{
				"-o", ifname,
				"-m", "conntrack", "--ctstate", "RELATED,ESTABLISHED", ---> this was second
				"-j", "ACCEPT",
			},
			protocol: protocol,
		},
		{
			table: "filter",
			chain: "FORWARD",
			args: []string{
				"-i", ifname, ---> this was first
				"-j", "ACCEPT",
			},
			protocol: protocol,
		},
```
and 
```
		{
			table: "nat",
			chain: "POSTROUTING",
			args: []string{
				"-s", masqueradeIP,
				"-j", "MASQUERADE", ---> this was second
			},
			protocol: protocol,
		},
		{
			table: "nat",
			chain: "POSTROUTING",
			args: []string{
				"-s", cidr.String(),
				"-j", "MASQUERADE", ---> this was first
			},
			protocol: protocol,
		},
```
?

I know that before we were inserting and now we are appending, but within these rules is there any reason for maintaining order?

won't the `-i` and `-o` matches just be complimentary to each other? same with `-s` match on cluster cidr versus masquerade... Not sure why these rules are been flipped :(

Reply:
honestly I didn't put too much thought into this :smile: 
it felt more natural for me to keep the actual order we had before instead of modifying the tests for the new order if I hadn't changed it here (i.e flipping the code and not the actual order in iptables) to not have to think if that order matters (complimentary/conflicting/etc..).

do you think it's better to do the opposite?

Reply:
oh lol! so its either we flip it here or need to change it in the tests ? haha, nvm its fine as it has 0 functionality impact.
As a reviewer it was just confusing to see code thrown around in these parts when it wasn't necessary or expected.